### DISCUSSION TITLE
Adjust docker container security to reduce lag

### DISCUSSION THREAD

**Discussion by suprnova74** [itzg/docker-minecraft-server#1448] [General]
### Enhancement Type

Not sure

### Describe the enhancement

I noticed a marked lag spike ~every 2 to 3 seconds or so for .25 to .5 seconds making fighting anything rather painful.  Upon searching some docker posts online, I found this: https://pythonspeed.com/articles/docker-performance-overhead/

I didn't track down the exact setting, but noticed when I completely disable docker security the lag is completely gone.  

It'd be nice to determine what setting(s) would help improve speed while still offering some of the security docker provides and have the process to spin up a server set that accordingly.

**Comment by itzg** [comment]
That's an interesting find, but I'm not sure what you want me to enhance? Even adding to the container's README is not prudent since the flag is a general docker parameter. 

Converted to discussion.

**Comment by suprnova74** [comment]
actually, upon re-examining, specifically I enabled privileged.  I guess could just run with --privileged versus editing the .json.  I'm unsure what permission minecraft inside docker is trying to hit to see if specific rights could be opened to alleviate the issue.  I'm guessing it's something to do with seccomp but that's a wild guess.

**Reply by itzg** [reply]
> actually, upon re-examining, specifically I enabled privileged. I guess could just run with --privileged versus editing the .json. 

Huh? What json are you editing to alter that?

> I'm unsure what permission minecraft inside docker is trying to hit to see if specific rights could be opened to alleviate the issue. I'm guessing it's something to do with seccomp but that's a wild guess.

I think it's just going to be a general Linux syscall access that bumps into that, like file access, etc.

**Comment by suprnova74** [comment]
my docker skills are rather rudimentary so wasn't sure how to run it privileged when (re)starting the container.  i think i had the order of params wrong so was throwing me off.  Anyway I edited  /var/lib/docker/containers/<containerid>/hostconfig.json.  found it after poking around the internet for a while.

It houses a parameter to run elevated.  Unfortunately it resets everytime so gets annoying.  it's also locked down so need to edit via sudo.
